src/pint/output/publish.py

```diff
@@ -0,0 +1,316 @@
+    else:
+        return f"{param.name}, {param.description}{publish_param_unit(param)}\dotfill &  {publish_param_value(param)} \\\\ \n"
+
+
+def publish(
+    model: TimingModel,
+    toas: TOAs,
+    include_dmx=False,
+    include_noise=False,
+    include_jumps=False,
+    include_zeros=False,
+    include_fd=False,
+    include_glitches=False,
+    include_swx=False,
+    include_tzr=False,
+):
+    """Generate LaTeX summary of a given timing model and TOAs.
+
+    Parameters
+    ----------
+    model: pint.model.timing_model.TimingModel
+        Input timing model
+    toas: TOAs
+        Input TOAs
+    include_dmx: bool
+        Whether to include DMX paremeters (default is False)
+    include_noise: bool
+        Whether to include noise paremeters (default is False)
+    include_jumps: bool
+        Whether to include jump paremeters (JUMPs, DMJUMPs) (default is False)
+    include_zeros: bool
+        Whether to include paremeters which are zero (default is False)
+    include_fd: bool
+        Whether to include FD paremeters (default is False)
+    include_glitches: bool
+        Whether to include glitch paremeters (default is False)
+    include_swx: bool
+        Whether to include SWX paremeters (default is False)
+    include_tzr: bool
+        Whether to include TZR paremeters (default is False)
+
+    Returns
+    -------
+    latex_summary: str
+        The LaTeX summary
+    """
+    mjds = toas.get_mjds()
+    mjd_start, mjd_end = int(min(mjds.value)), int(max(mjds.value))
+    data_span_yr = (mjd_end - mjd_start) / 365.25
+
+    fit_method = (
+        "GLS"
+        if np.any([nc.introduces_correlated_errors for nc in model.NoiseComponent_list])
+        else "WLS"
+    )
+
+    if toas.is_wideband():
+        res = WidebandTOAResiduals(toas, model)
+        toares = res.toa
+        dmres = res.dm
+    else:
+        res = Residuals(toas, model)
+        toares = res
+
+    exclude_params = [
+        "START",
+        "FINISH",
+        "NTOA",
+        "CHI2",
+        "DMDATA",
+        "PSR",
+        "EPHEM",
+        "CLOCK",
+        "UNITS",
+        "TIMEEPH",
+        "T2CMETHOD",
+        "DILATEFREQ",
+        "INFO",
+        "ECL",
+        "BINARY",
+    ]
+
+    exclude_components = [Wave]
+    if not include_dmx:
+        exclude_components.append(DispersionDMX)
+    if not include_jumps:
+        exclude_components.extend([PhaseJump, DispersionJump])
+    if not include_fd:
+        exclude_components.append(FD)
+    if not include_noise:
+        exclude_components.append(NoiseComponent)
+    if not include_glitches:
+        exclude_components.append(Glitch)
+    if not include_swx:
+        exclude_components.append(SolarWindDispersionX)
+    if not include_tzr:
+        exclude_components.append(AbsPhase)
+
+    with StringIO("w") as tex:
+        tex.write("\\documentclass{article}\n")
+        tex.write("\\begin{document}\n")
+
+        tex.write("\\begin{table}\n")
```

Comment:
Perhaps we should be using AASTex deluxetable environment?

Reply:
That requires importing a third-party sty file. I think the output should work on a minimal latex installation.